Cython/Compiler/Nodes.py

```diff
@@ -877,7 +877,8 @@ def analyse(self, env, nonempty=0, is_self_arg=False):
 
             # inject type declaration from annotations
             # this is called without 'env' by AdjustDefByDirectives transform before declaration analysis
-            if self.annotation and env and env.directives['annotation_typing'] and self.base_type.name is None:
+            if (self.annotation and env and env.directives['annotation_typing']
+                    and getattr(self.base_type, "name", None) is None):
```

Comment:
[Still not](https://github.com/cython/cython/pull/3372#discussion_r382061610) a friend of `getattr()`, in cases like this. What's the reason why there would be no `name`? (That's completely unclear from this use of `getattr()`.)

Reply:
`self.base_type` should apparently be a subclass of [`CBaseTypeNode`](https://github.com/cython/cython/blob/master/Cython/Compiler/Nodes.py#L927) (which doesn't have an attribute `name`). In practice it seems to be one of `CSimpleBaseTypeNode` (which has an attribute `name`), or `CAnalysedBaseTypeNode` (which doesn't actually inherit from `CBaseTypeNode`, and which doesn't have an attribute `name`).

`getattr` seemed the easiest way out of that mess, but if you want to propose an alternative I'd be happy to use it instead.

Reply:
See? You actually know a lot more about the underlying behaviour than the code originally allowed readers to infer. Thanks for the comment.